txn.go

```diff
@@ -172,6 +172,60 @@ func (txn *Txn) Commit() {
 	}
 }
 
+// CreateIndexes creates the given indexes for each value of the root tree
+func (txn *Txn) CreateIndexes(table string, indexes ...*IndexSchema) error {
+	if !txn.write {
+		return fmt.Errorf("cannot create index in read-only transaction")
+	}
+
+	// Get the table schema
+	tableSchema, ok := txn.db.schema.Tables[table]
+	if !ok {
+		return fmt.Errorf("invalid table '%s'", table)
+	}
+
+	// and the id indexer
+	idSchema := tableSchema.Indexes[id]
+	idIndexer := idSchema.Indexer.(SingleIndexer)
+
+	// create iterator over all entries
+	indexIterator, _, err := txn.getIndexIterator(table, id)
+	if err != nil {
+		return err
+	}
+	iter := &radixIterator{iter: indexIterator}
+
+	// create all indexes for each index entry
+	for next := iter.Next(); next != nil; next = iter.Next() {
+
+		// Get the primary ID of the object
+		ok, idVal, err := idIndexer.FromObject(next)
+		if err != nil {
+			return fmt.Errorf("failed to build primary index: %v", err)
+		}
+		if !ok {
+			return fmt.Errorf("object missing primary index")
+		}
+
+		for _, indexSchema := range indexes {
+			indexTxn := txn.writableIndex(table, indexSchema.Name)
+
+			var vals [][]byte
+			vals, err = txn.createIndexValue(indexSchema, idVal, next)
+			if err != nil {
+				return err
+			}
+
+			// Update the value of the index
+			for _, val := range vals {
+				indexTxn.Insert(val, next)
+			}
+		}
```

Comment:
I see you factored out `txn.createIndexValue` from `Insert` rather than copying it which makes sense, but unless I'm missing something, isn't the whole of this block identical to the `Insert` code? Could we factor out the whole thing? Or even just call `txn.Insert` directly? I guess that does slightly more work by replacing the ID index with existing values too which is a no-op but wasted effort.

I'd be interested to know if you measure significnatly worse times from just calling `Insert` for your use-case. The benefit is reduced duplication and no change to the current code path every other caller of this library is using. The change is straightforward so it's not high risk, but it's more than no change!